// tonic-io/lib/Girder/_webpack.config.js
var HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
    plugins: [ new HtmlWebpackPlugin() ],
    entry: [
        './lib/Girder/index.js',
        './lib/Girder/services',
    ],
    output: {
        path: './dist/girder',
        filename: 'girder-client.js',
    },
    module: {
        preLoaders: [
            {
                test: /\.js$/,
                loader: "eslint-loader",
                exclude: /node_modules/,
            },
        ],
        loaders: [
            { test: require.resolve("./services"), loader: "expose?Girder.services" },
            { test: require.resolve("./index.js"), loader: "expose?Girder" },
            { test: /\.js$/, exclude: /node_modules/, loader: "babel?presets[]=react,presets[]=es2015" },
        ],
    },
    eslint: {
        configFile: '.eslintrc',
    },
    devServer: {
        contentBase: './dist/girder',
        port: 9999,
        hot: true,
        quiet: false,
        noInfo: false,
        stats: {
            colors: true,
        },
        proxy: {
            '/api*': 'http://localhost:8080',
            '/static/*': 'http://localhost:8080',
        },
    },
};
