// paraviewweb/src/React/CollapsibleControls/PlotlyChartControl/Histogram2D.js
import React                    from 'react';
import style from 'PVWStyle/ReactWidgets/PlotlySelectionWidgets.mcss';
import Select from 'react-select';
import 'react-select/dist/react-select.css';


const PlotlyHistogram2DWidget = (props) => {
  function handleChange(event) {
    // const rootContainer = event.target.parentNode.parentNode.parentNode;
    // const newXArray = rootContainer.querySelector('.jsX').value;
    // const newYArray = rootContainer.querySelector('.jsY').value;
    // const forceNewPlot = props.arrays[props.chartState.x] !== props.arrays[newXArray] ||
    //   props.arrays[props.chartState.y] !== props.arrays[newYArray];

    // have to update the correct variable (was x or y variable changed?)
    let newXArray = props.chartState.x, 
        newYArray = props.chartState.y;

    if (event.axis === 'x') {
      newXArray = event.value;
    }

    if (event.axis === 'y') {
      newYArray = event.value;
    }


    props.onChange({
      chartType: 'Histogram2D',
      x: newXArray,
      y: newYArray,
      forceNewPlot: true,
    });
  }
              // <select className={['jsX', style.fullWidth].join(' ')} onChange={handleChange} value={props.chartState.x}>
              //   {Object.keys(props.arrays).map(arrayName => <option value={arrayName} key={arrayName}>{arrayName}</option>)}
              // </select>

              // <select className={['jsY', style.fullWidth].join(' ')} onChange={handleChange} value={props.chartState.y}>
              //   {Object.keys(props.arrays).map(arrayName => <option value={arrayName} key={arrayName}>{arrayName}</option>)}
              // </select>              
  return (
    <div>
      <table className={style.fullWidth}>
        <tbody>
          <tr>
            <td className={style.label}>x</td>

            <td className={style.fullWidth}>

              <Select className={['jsX', style.fullWidth].join(' ')}
                  clearable="false"
                  value={props.chartState.x}
                  options={Object.keys(props.arrays).map(name => {return {value: name, label: name, axis: 'x'};} )}
                  onChange={handleChange}
              />


            </td>
          </tr>
          <tr>
            <td className={style.label}>y</td>
            <td>

              <Select className={['jsY', style.fullWidth].join(' ')}
                  clearable="false"
                  value={props.chartState.y}
                  options={Object.keys(props.arrays).map(name => {return {value: name, label: name, axis: 'y'};} )}
                  onChange={handleChange}
              />


            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

PlotlyHistogram2DWidget.propTypes = {
  chartState: React.PropTypes.object,
  arrays: React.PropTypes.object,
  onChange: React.PropTypes.func,
};

PlotlyHistogram2DWidget.defaultProps = {
  chartState: {},
  arrays: [],
  onChange: () => {},
};

export default PlotlyHistogram2DWidget;
