// tonic-io/webpack.config.js
module.exports = {
    plugins: [],
    entry: './lib/index.js',
    output: {
        path: './dist',
        filename: 'Tonic-io.js',
    },
    module: {
        preLoaders: [
            {
                test: /\.js$/,
                loader: "eslint-loader",
                exclude: /node_modules/,
            },
        ],
        loaders: [
            {
                test: require.resolve("./lib/index.js"),
                loader: "expose?Tonic.io",
            }, {
                test: /\.json$/,
                loader: 'json-loader',
            }, {
                test: /\.js$/,
                include: /node_modules\/tonic-/,
                loader: "babel?presets[]=react,presets[]=es2015",
            }, {
                test: /\.js$/,
                exclude: /node_modules/,
                loader: "babel?presets[]=react,presets[]=es2015",
            },
        ],
    },
    eslint: {
        configFile: '.eslintrc',
    },
    devServer: {
        contentBase: './dist/',
        port: 9999,
        hot: true,
        quiet: false,
        noInfo: false,
        stats: {
            colors: true,
        },
        proxy: {
            '/api*': 'http://localhost:8080',
            '/static/*': 'http://localhost:8080',
        },
    },
};
