// paraviewweb/src/Rendering/Chart/PlotlyChartBuilder/Histogram2D.js

export default function Histogram2D(chartState, csvReader, arraysInfo) {
  const arrayNames = Object.keys(arraysInfo);
  if (!chartState.x) {
    chartState.x = arrayNames[0];
  }

  if (!chartState.y) {
    chartState.y = arrayNames[(arrayNames.length >= 2 ? 1 : 0)];
  }

  return {
    forceNewPlot: true, // Due to a bug in plotly redraw w/ 2D histogram
    traces: [
      {
        x: csvReader.getColumn(chartState.x),
        y: csvReader.getColumn(chartState.y),
        type: 'histogram2d',
        nbinsx: 32,
        nbinsy: 32,
      },
    ],
    layout: {
      xaxis: {
        title: chartState.x,
      },
      yaxis: {
        title: chartState.y,
      },
    },
  };
}

// old, original code that allowed for only traces to be returned, not all chart
// data
//   return [
//     {
//       x: csvReader.getColumn(chartState.x),
//       y: csvReader.getColumn(chartState.y),
//       type: 'histogram2d',
//       forceNewPlot: true,      // Due to a bug in plotly redraw w/ 2D histogram
//     },
//   ];
// }
